recipes/apr-util/all/conanfile.py

```diff
@@ -0,0 +1,204 @@
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+        "with_nss": [True, False],
+        "with_commoncrypto": [True, False],
+        "dbm": [False, "gdbm", "ndbm", "db"],
+        "with_expat": [True, False],
+        "with_mysql": [True, False],
+        "with_postgresql": [True, False],
+        "with_sqlite3": [True, False],
+        "with_lber": [True, False],
+        "with_ldap": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": False,
+        "with_nss": False,
+        "with_commoncrypto": False,
+        "dbm": False,
+        "with_expat": True,
+        "with_mysql": False,
+        "with_postgresql": False,
+        "with_sqlite3": False,
+        "with_lber": False,
+        "with_ldap": False,
+    }
+
+    _autotools = None
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+        if not self.options.with_expat:
+            raise ConanInvalidConfiguration("expat cannot be disabled (at this time) (check back later)")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        self.requires("apr/1.7.0")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1g")
+        if self.options.with_nss:
+            # self.requires("nss/x.y.z")
+            raise ConanInvalidConfiguration("CCI has no nss recipe (yet)")
+        if self.options.with_commoncrypto:
+            # self.requires("commoncrypto/x.y.z")
+            raise ConanInvalidConfiguration("CCI has no commoncrypto recipe (yet)")
+        if self.options.dbm == "gdbm":
+            # self.requires("gdbm/x.y.z")
+            raise ConanInvalidConfiguration("CCI has no gdbm recipe (yet)")
+        elif self.options.dbm == "ndbm":
+            # self.requires("ndbm/x.y.z")
+            raise ConanInvalidConfiguration("CCI has no ndbm recipe (yet)")
+        elif self.options.dbm == "db":
+            # self.requires("libdb/x.y.z")
+            raise ConanInvalidConfiguration("CCI has no libdb recipe (yet)")
+        if self.options.with_lber:
+            # self.requires("lber/x.y.z")
+            raise ConanInvalidConfiguration("CCI has no lber recipe (yet)")
+        if self.options.with_ldap:
+            # self.requires("ldap/x.y.z")
+            raise ConanInvalidConfiguration("CCI has no ldap recipe (yet)")
```

Comment:
IIRC we have ldap in Bincrafters ...

Reply:
I can't find any reference to it.
So it's a FIXME for later, I guess?

Reply:
good idea